src/components/images/imageLoader.js

```diff
@@ -1,50 +1,83 @@
-define(['lazyLoader', 'imageFetcher', 'layoutManager', 'browser', 'appSettings', 'userSettings', 'require', 'css!./style'], function (lazyLoader, imageFetcher, layoutManager, browser, appSettings, userSettings, require) {
-    'use strict';
+import * as lazyLoader from 'lazyLoader';
+import * as userSettings from 'userSettings';
+import 'css!./style';
+/* eslint-disable indent */
 
-    var self = {};
+    export function lazyImage(elem, source = elem.getAttribute('data-src')) {
+        if (!source) {
+            return;
+        }
 
-    function fillImage(elem, source, enableEffects) {
+        fillImageElement(elem, source);
+    }
 
-        if (!elem) {
-            throw new Error('elem cannot be null');
+    export function fillImage(entry) {
+        if (!entry) {
+            throw new Error('entry cannot be null');
         }
 
-        if (!source) {
-            source = elem.getAttribute('data-src');
+        var source = undefined;
+        if (entry.target) {
+            source = entry.target.getAttribute('data-src');
+        } else {
+            source = entry;
         }
 
-        if (!source) {
-            return;
+        if (entry.intersectionRatio > 0 && source) {
+            fillImageElement(entry.target, source);
+        } else if (!source) {
+            emptyImageElement(entry.target);
         }
```

Comment:
I played with it on TV layout.

- Open Music / Albums (may be any big library).
- Scroll down using keyboard.
- Then scroll up step by step.

Cards loose background. For some reason, they have `background` being out of view (looked in Inspector). Maybe loading is not fast enough for scrolling. And you make them empty because `source = entry.target.getAttribute('data-src')` is `undefined`.

Reply:
This will fix, but no fading effect: just show image if it is in view, regardless of value of `data-src` attribute.
```js
        if (entry.intersectionRatio > 0) {
            if (source) fillImageElement(entry.target, source);
        } else if (!source) {
            emptyImageElement(entry.target);
        }
```

Reply:
I tried something, it seems to somewhat fix it for me. I have a few cases where images take a while to load, so the cards show up empty&& entry.intersectionRatio === 0 (But that happens with the current implementation as well).